addons/mail/models/mail_template.py

```diff
@@ -237,6 +237,35 @@ def onchange_sub_model_object_value_field(self):
             self.sub_model_object_field = False
             self.null_value = False
 
+    @api.multi
+    def write(self, vals):
+        # This override is for checking that the template is correct when
+        # saving it
+        if vals.get('body_html'):
+            # render_template can return empty results if the mako rendering
+            # crashes or if the body_html is empty, therefore we will only perform
+            # this check if the new body_html is non-empty
+            if vals.get('model_id'):
+                _model = self.env['ir.model'].browse(vals['model_id']).model
+            else:
+                _model = self.model
+
+            Model = self.env[_model]
+
+            if not Model._abstract:
+                # Get the latest record so as to not use older, potentially
+                # deprecated/obsolete records
+                res_ids = Model.search([], limit=1, order='id DESC')
+                res_ids = res_ids[0].id if res_ids else None
```

Comment:
res_ids = Model.search(..).ids[:1] and later, if res_ids (because you will have either a void list, either a list with one element).